src/main/java/org/opensearch/flowframework/FlowFrameworkPlugin.java

```diff
@@ -48,9 +46,9 @@ public Collection<Object> createComponents(
         IndexNameExpressionResolver indexNameExpressionResolver,
         Supplier<RepositoriesService> repositoriesServiceSupplier
     ) {
-        this.client = client;
-        CreateIngestPipelineStep createIngestPipelineStep = new CreateIngestPipelineStep(client);
-        CreateIndexStep createIndexStep = new CreateIndexStep(client);
-        return ImmutableList.of(createIngestPipelineStep, createIndexStep);
+        WorkflowStepFactory workflowStepFactory = WorkflowStepFactory.create(client);
+        WorkflowProcessSorter workflowProcessSorter = WorkflowProcessSorter.create(workflowStepFactory, threadPool.generic());
```

Comment:
I agree, we should pass threadpool, or better yet just use the injected version of it in a transport action.  I briefly went down that road but in the interest of getting the bug (#50) fixed it turned out doing tests with the thread pool version was far more complicated than I wanted to deal with, as I think that'll require IT rather than unit test.

Reply:
Are you planning to take care of passing threadpool in #50 ?